### PR TITLE
Make file upload question optional

### PR DESCRIPTION
**Pull request by hmhsu**
### Description
Add a skip button to fileupload question

Screen Shot 2021-06-21 at 3 32 28 PM

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)
#1210

### CODE REVIEW

File: universal-application-tool-0.0.1/app/controllers/applicant/ApplicantProgramBlocksController.java [hunk 1]
```diff
@@ -208,6 +208,60 @@ public ApplicantProgramBlocksController(
...
+   * empty file key which marks the file quesiton as seen but not answered and redirects the user to
+   * the next incomplete block.
+   */
+  @Secure
+  public CompletionStage<Result> skipFile(
+      Request request, long applicantId, long programId, String blockId, boolean inReview) {
+    CompletionStage<String> applicantStage = this.applicantService.getName(applicantId);
+
+    return applicantStage
+        .thenComposeAsync(v -> checkApplicantAuthorization(profileUtils, request, applicantId))
```
Comment (bion) [inline universal-application-tool-0.0.1/app/controllers/applicant/ApplicantProgramBlocksController.java:223]: suggestion (blocking): add `httpExecutionContext.current()`

File: universal-application-tool-0.0.1/app/controllers/applicant/ApplicantProgramBlocksController.java [hunk 2]
```diff
@@ -208,6 +208,60 @@ public ApplicantProgramBlocksController(
...
 
+  /**
+   * This method is used by the file upload question. If an user chooses to skip uploading a file,
+   * they click the skip button on the page which takes them to this endpoint. This method puts an
+   * empty file key which marks the file quesiton as seen but not answered and redirects the user to
+   * the next incomplete block.
+   */
+  @Secure
+  public CompletionStage<Result> skipFile(
+      Request request, long applicantId, long programId, String blockId, boolean inReview) {
```
Comment (bion) [inline universal-application-tool-0.0.1/app/controllers/applicant/ApplicantProgramBlocksController.java:219]: suggestion (blocking): please test this method in `ApplicantProgramBlocksControllerTest`

File: universal-application-tool-0.0.1/app/controllers/applicant/ApplicantProgramBlocksController.java (hunk 1 above)
Comment (hmhsu) [reply universal-application-tool-0.0.1/app/controllers/applicant/ApplicantProgramBlocksController.java]: Done.

File: universal-application-tool-0.0.1/app/controllers/applicant/ApplicantProgramBlocksController.java (hunk 2 above)
Comment (hmhsu) [reply universal-application-tool-0.0.1/app/controllers/applicant/ApplicantProgramBlocksController.java:223]: Done.

**Review by bion — APPROVED** [review]

**Pull request merged**